### PR TITLE
Added Roslyn Completion Service

### PR DESCRIPTION
**Pull request by OmarTawfik**
@Microsoft/visualfsharp 

In this PR:

- Added Completion Service
- Fixed tests to run against Roslyn version of VS binaries

### CODE REVIEW

**Comment by Pilchie** [comment]
Also tagging @dotnet/roslyn-ide to take a look (@cyrusnajmabadi, @jasonmalinowski, @rchande, @mattwar)

File: vsintegration/src/FSharp.Editor/CompletionProvider.fs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Tagging
+open Microsoft.VisualStudio.Shell
+open Microsoft.VisualStudio.Shell.Interop
+
+open Microsoft.FSharp.Compiler.Parser
+open Microsoft.FSharp.Compiler.Range
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+type internal FSharpCompletionProvider(workspace: Workspace, serviceProvider: SVsServiceProvider) =
```
Comment (mattwar) [inline vsintegration/src/FSharp.Editor/CompletionProvider.fs]: >serviceProvider: SVsServiceProvider [](start = 61, length = 35)

This is sort of a layering violation.   Is there no other way to implement this without accessing other VS layer services?

Comment (OmarTawfik) [inline vsintegration/src/FSharp.Editor/CompletionProvider.fs]: AFAIK, no, in the existing implementation (see [XmlDocumentation.fs](https://github.com/Microsoft/visualfsharp/blob/master/vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs)). Can you please explain?

Comment (DustinCampbell) [inline vsintegration/src/FSharp.Editor/CompletionProvider.fs]: Is the IVsXMLMemberIndexService the only reason for the layer violation? Seems like an abstraction could be added to solve this.

Comment (mattwar) [inline vsintegration/src/FSharp.Editor/CompletionProvider.fs]: The general idea for these providers is that they are not dependent on VS, so they can work outside of VS, either to run out of proc or be used within a different host.

Comment (OmarTawfik) [inline vsintegration/src/FSharp.Editor/CompletionProvider.fs]: Yes. We can log a future work item for redesigning this.

**Comment by OmarTawfik** [comment]
@dotnet-bot test this please

File: vsintegration/src/FSharp.Editor/CompletionProvider.fs (hunk 1 above)
Comment (DustinCampbell) [inline vsintegration/src/FSharp.Editor/CompletionProvider.fs]: Yup. @OmarTawfik and I chatted about that easier. For now, this is VS-only, and we can factor it differently after this goes in.

Comment (mattwar) [inline vsintegration/src/FSharp.Editor/CompletionProvider.fs]: okay, that was my only hesitation.

**Comment by CyrusNajmabadi** [comment]
Looks ok to me.

**Pull request merged**